Description: Port missing collate feature and flag from glabels-3 to batch
 While glabels-3 contains collate feature to collate multiple labels
 the same feature is missing from the batch binary. Added as a new flag
 which enables the same feature as the collate checkbox in the GUI.
 .
 glabels (3.4.1-4.2) UNRELEASED; urgency=medium
 .
   [ Martin Brozkeff Malec ]
   * Non-maintainer upload.
   * Applied collate patch for glabels-batch
Author: Martin Brozkeff Malec <martin@brozkeff.net>

---
The information above should follow the Patch Tagging Guidelines, please
checkout https://dep.debian.net/deps/dep3/ to learn about the format. Here
are templates for supplementary fields that you might want to add:

Origin: (upstream|backport|vendor|other), (<patch-url>|commit:<commit-id>)
Bug: <upstream-bugtracker-url>
Bug-Debian: https://bugs.debian.org/<bugnumber>
Bug-Ubuntu: https://launchpad.net/bugs/<bugnumber>
Forwarded: (no|not-needed|<patch-forwarded-url>)
Applied-Upstream: <version>, (<commit-url>|commit:<commid-id>)
Reviewed-By: <name and email of someone who approved/reviewed the patch>
Last-Update: 2024-04-19

--- glabels-3.4.1.orig/src/glabels-batch.c
+++ glabels-3.4.1/src/glabels-batch.c
@@ -44,6 +44,7 @@ static gint     n_sheets         = 1;
 static gint     first            = 1;
 static gboolean outline_flag     = FALSE;
 static gboolean reverse_flag     = FALSE;
+static gboolean collate_flag     = FALSE;
 static gboolean crop_marks_flag  = FALSE;
 static gchar    *input           = NULL;
 static gchar    **remaining_args = NULL;
@@ -61,6 +62,8 @@ static GOptionEntry option_entries[] = {
          N_("print outlines (to test printer alignment)"), NULL},
         {"reverse", 'r', 0, G_OPTION_ARG_NONE, &reverse_flag,
          N_("print in reverse (i.e. a mirror image)"), NULL},
+        {"collate", 'a', 0, G_OPTION_ARG_NONE, &collate_flag,
+         N_("collate if more copies are present (e.g. to print same label on one page)"), NULL},
         {"cropmarks", 'C', 0, G_OPTION_ARG_NONE, &crop_marks_flag,
          N_("print crop marks"), NULL},
         {"input", 'i', 0, G_OPTION_ARG_STRING, &input,
@@ -162,6 +165,7 @@ main (int argc, char **argv)
                         gl_print_op_set_first           (print_op, first);
                         gl_print_op_set_outline_flag    (print_op, outline_flag);
                         gl_print_op_set_reverse_flag    (print_op, reverse_flag);
+                        gl_print_op_set_collate_flag    (print_op, collate_flag);
                         gl_print_op_set_crop_marks_flag (print_op, crop_marks_flag);
                         if (merge)
                         {
